src/main/perl/Filesystem.pm

```diff
@@ -157,6 +157,66 @@ sub remove_if_needed
 
 =pod
 
+=head2 check_in_fstab
+
+Parse the fstab entry for the C<$self> filesystem. When a (PART)UUID is found, 
+this is used instead of the device name or label. 
+If it concerns a protected (i.e. never-modify) mountpoint or filesystem type,
+it will add it, but changes will not be allowed.
+When the entry should be added or inserted, this sub will return the device name to use,
+otherwise undef is returned.
+
+=cut
+
+sub check_in_fstab
+{
+    my ($self, $fh, $protected) = @_;
+    my $add = 1;
+    my ($ndevice, $otype);
+    my $txt = "$fh";
+    my $re = qr!^\s*([^#\s]\S+)\s+$self->{mountpoint}\/?\s+(\S+)\s!m;
+    my $devpath = $self->{block_device}->devpath;
+    while ($txt =~ m/$re/mg) {
+        $add = 0;
+        (my $device, $otype) = ($1, $2);
+        if ($device =~ m/^(PART)?UUID=(\S+)/) {
+            my ($type_uuid, $fstab_uuid) = ($1 || '' , $2);
```

Comment:
the typeuuid looks odd.
```perl
if ($device =~ m/^((?:PART)?UUID)=(\S+)/) {
    my ($type_uuid, $fstab_uuid) = ($1 , $2);
}
```
and replace all `${type_uuid}UUID` with `${type_uuid}`